codec/src/main/java/io/netty/handler/internal/svm/codec/compression/ZlibCodecFactorySubstitutions.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.internal.svm.codec.compression;
+
+import io.netty.handler.codec.compression.JdkZlibDecoder;
+import io.netty.handler.codec.compression.JdkZlibEncoder;
+import io.netty.handler.codec.compression.ZlibDecoder;
+import io.netty.handler.codec.compression.ZlibEncoder;
+import io.netty.handler.codec.compression.ZlibWrapper;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+/**
+ * This substitution avoid having jcraft zlib added to the build
```

Comment:
Just from curiosity, what is the problem with `jcraft` lib? Also, could we simply support two different implementations in `io.netty.handler.codec.compression.ZlibCodecFactory` based on a config flag. That way we would avoid the substitutions--I would use substitutions for temporary support of external projects. Since we are in the same project, I would simply implement it in netty.